app/models/user.rb

```diff
@@ -0,0 +1,50 @@
+class User < ApplicationRecord
+  has_many :votes
+  has_many :works, through: :votes
+
+  validates :username, presence: true, uniqueness: true
+
+  def self.recommendations(new_user)
+    all_users = User.where.not(id: nil)
+
+    user_indices = Hash.new
+    all_users.each do |user|
+      user_indices[user] = User.calculate_jaccard(new_user, user)
+    end
+
+    user_indices = user_indices.sort_by { |user, index| index }
+    top_similar_raw = user_indices.last(3)
+
+    top_similar = Array.new
+    top_similar_raw.each do |user_index|
+      top_similar << user_index[0]
+    end
+
+    recommendations = User.get_recommendations(top_similar, new_user) # this should probably not be a class method but it is for now for testing
+
+    return recommendations.empty? ? "no recommendations at this time" : recommendations.uniq
+  end
```

Comment:
The condition in line 25 is display logic, and would probably fit better in the view.